### DISCUSSION TITLE
[Ent] Let a Sidekiq instance opt-out from becoming the senate leader.

### DISCUSSION THREAD

**Discussion by nasa42** [sidekiq/sidekiq#5415] [Ideas]
Due to legacy reasons, we have a shared Redis database between two separate Rails apps (app A & app B). Only app A has definitions for the periodic jobs. Everything works great if a Sidekiq instance running app A is elected as the leader of the senate. However, if a Sidekiq instance running app B is elected as the leader, periodic jobs stop working (as app B doesn't define periodic jobs).
We've fixed this temporarily by patching `Sidekiq::Senate#stage_coup!` to always return `false` in app B, however a config option to let a Sidekiq instance just opt out of becoming leader would be nice.

This has been discussed in past (https://github.com/mperham/sidekiq/issues/4078 and https://github.com/mperham/sidekiq/issues/3771) with a conclusion that sharing a Redis db between two different apps is a bad idea. The scope of this thread is to find out if it would be feasible to just add an option to let a Sidekiq worker opt out of becoming the leader.

**Comment by mperham** [comment]
I've never wanted to do this because of the worst case: what happens when you accidentally configure all processes to opt-out? It also, as you note, allows you to workaround Sidekiq's "one Redis per app" rule which is not a good reason to do it.
